File: 11.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        i, j = 0, len(height) -1
        C = 0
        while i < j:
            C = max(C, (j - i) * min(height[i], height[j]))
            if height[i] < height[j]:
                i += 1
            else:
                j -= 1
        
        return C

File: 11.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int h=0;
        int C=0;
        while(1){
            int l=0;
            int r=height.size()-1;
            while(height[l] < h) l++;
            while(height[r] < h) r--;
            C=max(C, (r - l ) * h);
            if(l >= r) break;
            h++;
        }
        return C;
    }
};
